test/mitmproxy/tools/console/test_statusbar.py

```diff
@@ -31,3 +34,29 @@ def test_statusbar(monkeypatch):
 
     bar = statusbar.StatusBar(m)  # this already causes a redraw
     assert bar.ib._w
+
+
+@pytest.mark.parametrize("message,ready_message", [
+    ("", [(None, ""), ("warn", "")]),
+    (("info", "Line fits into statusbar"), [("info", "Line fits into statusbar"),
+                                            ("warn", "")]),
+    ("Line doesn't fit into statusbar", [(None, "Line does..."),
+                                         ("warn", "(more in eventlog)")]),
+    (("alert", "Two lines.\nFirst fits"), [("alert", "Two lines."),
+                                           ("warn", "(more in eventlog)")]),
+    ("Two long lines\nFirst doesn't fit", [(None, "Two long ..."),
+                                           ("warn", "(more in eventlog)")])
+])
+def test_prep_message(message, ready_message):
+    m = mock.Mock()
```

Comment:
We can avoid mocking here if you move the `self.master.ui.get_cols_rows()` call into sig_message and pass it as a `max_width` argument.